### PR TITLE
Add blog post about StarlingX security features

### PR DESCRIPTION
**Pull request by ildikov**
This patch adds an article by Ian Jolliffe about security related
functionality in StarlingX.

Signed-off-by: Ildiko Vancsa <ildiko.vancsa@gmail.com>

### CODE REVIEW

File: site/blog/starlingx-release-2-security-features.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+- **UEFI secure boot**: Can be used to securely boot from the lowest levels of the stack such that the operating system can be booted from BIOS to GRUB in a secure way.
+- **TPM device enablement**: StarlingX provides the ability to leverage a TPM device when it is available on a server. TPM devices can be used to store secrets, such as private keys for HTTPS. Once they are in the TPM device, the secret can only be read by authorized users and can never be modified. Another important use of TPM devices is to leverage entropy built into the device to ensure more randomness of any generated keys. 
+- **Certificate management**: In a Kubernetes environment the number of certificates can easily explode. We have some initial capabilities with StarlingX Release 2.0 where these certificates are managed more manually than we would like. In future releases, we are looking to collaborate with other projects to make it easier to manage these.
```
Comment (ian-jolliffewr) [inline site/blog/starlingx-release-2-security-features.md]: In future releases, we are looking to collaborate with other projects to make it easier to manage these certificates.

File: site/blog/starlingx-release-2-security-features.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+
+As we drive compute to the edge of the network many things change and become more interesting and new problems need to be solved. At a minimum the challenges include space and power limitations which then turns into balancing server costs while enabling emerging applications and new transformational use cases at the edge when you take the next steps. However; even though the way compute is delivered at the edge is different, the need for security remains constant or even more critical.
+
+Security is always an important consideration in any deployment. It gets more interesting when we no longer have the high level of physical security in the data center where there are retinal scanners at the doors. As we move to the edge we have an environment where there may be less physical security and nodes that are geographically distributed also make maintenance activities difficult.
+
+Within the StarlingX community these are challenges we have recognized and have been working on for a while to provide solutions for. The key approaches start with security enabled at the lowest level of the stack and build up from there. The key challenges we have tackled so far are:
```
Comment (ian-jolliffewr) [inline site/blog/starlingx-release-2-security-features.md]: Within the StarlingX community these are challenges we have recognized and have been working on for a while to provide solutions for.

I checked the wording below with help from Grammarly :)
Within the StarlingX community, these are challenges we have recognized and have been working on solutions for a while.

File: site/blog/starlingx-release-2-security-features.md [hunk 3]
```diff
@@ -0,0 +1,23 @@
+---
+title:  Securing the Edge with StarlingX
+author: ian-jolliffe
+date: 2019-10-30T01:32:05.627Z
+category: features-and-updates
+---
+Security on the edge is crucial. Learn more about how the StarlingX 2.0 release helps addressing this challenge. <!-- more -->
+
+
+As we drive compute to the edge of the network many things change and become more interesting and new problems need to be solved. At a minimum the challenges include space and power limitations which then turns into balancing server costs while enabling emerging applications and new transformational use cases at the edge when you take the next steps. However; even though the way compute is delivered at the edge is different, the need for security remains constant or even more critical.
```
Comment (ian-jolliffewr) [inline site/blog/starlingx-release-2-security-features.md]: At a minimum, the challenges include space and power limitations.  The next level of challenge includes balancing server costs while enabling emerging applications and new transformational use cases at the edge when you take the next steps.

Comment (ildikov) [reply site/blog/starlingx-release-2-security-features.md]: I added the changes to eliminate using 'challenge include' twice and the 'next level of challenge' sounded a bit odd to me.

File: site/blog/starlingx-release-2-security-features.md (hunk 2 above)
Comment (ildikov) [reply site/blog/starlingx-release-2-security-features.md]: The 'for a while' ending makes it sound somehow less appealing. Changed it to the suggested variation, can always update later. :)

File: site/blog/starlingx-release-2-security-features.md [hunk 4]
```diff
@@ -7,15 +7,15 @@ category: features-and-updates
...
 
 Security is always an important consideration in any deployment. It gets more interesting when we no longer have the high level of physical security in the data center where there are retinal scanners at the doors. As we move to the edge we have an environment where there may be less physical security and nodes that are geographically distributed also make maintenance activities difficult.
 
-Within the StarlingX community these are challenges we have recognized and have been working on for a while to provide solutions for. The key approaches start with security enabled at the lowest level of the stack and build up from there. The key challenges we have tackled so far are:
+Within the StarlingX community, these are challenges we have recognized and have been working on solutions for a while. The key approaches start with security enabled at the lowest level of the stack and build up from there. The key challenges we have tackled so far are:
```
Comment (ian-jolliffewr) [inline site/blog/starlingx-release-2-security-features.md]: Let's drop the "for a while" and then I think we are good.

Comment (ildikov) [reply site/blog/starlingx-release-2-security-features.md]: Done

File: site/blog/starlingx-release-2-security-features.md [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+- **Certificate management**: In a Kubernetes environment the number of certificates can easily explode. We have some initial capabilities with StarlingX Release 2.0 where these certificates are managed more manually than we would like. In future releases, we are looking to collaborate with other projects to make it easier to manage these certificates.
+- **Container isolation**:Fundamentally, with containers we are sharing a kernel. Cgroups and namespaces are the key enablers of containers and great capabilities to provide isolation of these workloads. However; there are some situations that require technologies that can help address the risks of a shared kernel. [Kata Containers](https://katacontainers.io), another project supported by the Open Infrastructure Foundation provides an excellent way to provide additional isolation and many of the benefits of containers. This community is making great strides with adoption by providing a solution that is a fusion between virtualization and container technologies. In StarlingX, we already have a specification to add support for Kata Containers and we expect this capability to land in a future release.
```
Comment (allisonprice) [inline site/blog/starlingx-release-2-security-features.md]: this should say OpenStack Foundation, not Open Infrastructure Foundation

**Pull request merged**